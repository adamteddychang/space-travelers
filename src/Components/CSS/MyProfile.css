html {
  overflow-y: scroll;
}

.Profile-main {
  display: flex;
  width: 90%;
  margin: 1rem auto;
  justify-content: space-around;

  /* border: 1px solid black; */
}

.My-Missions,
.My-Rockets {
  /* border: 1px solid black; */
  padding: 0 4rem 4rem;
}

.My-Missions h2,
.My-Rockets h2 {
  margin: 1rem;
  font-size: 40px;
  font-weight: 600;
}

.ul-contain-miss {
  padding: 0 0.5rem;
  list-style-type: none;
  border: 1px solid black;
}

.ul-contain-miss li {
  margin: 1rem 0;
  padding: 1rem 1.5rem;
  border-bottom: 1px solid black;
  font-size: 20px;
}

.ul-contain-miss li:last-child {
  border: none;
}
