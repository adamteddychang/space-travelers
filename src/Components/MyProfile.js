import React from 'react';

function MyProfile() {
  return (
    <div>
      <h2>Under Construction My Profile</h2>
    </div>
  );
}

export default MyProfile;
