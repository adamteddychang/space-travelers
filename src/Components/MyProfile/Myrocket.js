import React from 'react';
import { useSelector } from 'react-redux';

const Myrocket = () => {
  const myRockets = useSelector((state) => state.rockets.filter((rocket) => rocket.reserved));
  return (
    <div>
      <h2>My Missions</h2>
      {
        (myRockets.length === 0)
          ? <p>Empty</p>
          : (
            <ul>
              {
                myRockets.map((rocket) => (
                  <li key={rocket.rocket_id}>
                    {rocket.rocket_name}
                  </li>
                ))
              }
            </ul>
          )
      }
    </div>
  );
};

export default Myrocket;
