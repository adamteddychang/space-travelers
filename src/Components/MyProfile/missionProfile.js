import React from 'react';
import { useSelector } from 'react-redux';

const MyMissions = () => {
  const myMissions = useSelector((state) => state.missions.filter((mission) => mission.joined));
  return (
    <div className="My-Missions">
      <h2>My Missions</h2>
      {
        (myMissions.length === 0)
          ? (
            <ul className="ul-contain-miss">
              <li>Empty</li>
            </ul>
          )
          : (
            <ul className="ul-contain-miss">
              {
                myMissions.map((mission) => (
                  <li key={mission.mission_id}>
                    {mission.mission_name}
                  </li>
                ))
              }
            </ul>
          )
      }
    </div>
  );
};

export default MyMissions;
