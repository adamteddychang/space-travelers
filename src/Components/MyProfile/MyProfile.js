import React from 'react';
import MyMissions from './missionProfile';
import Myrocket from './Myrocket';

function MyProfile() {
  return (
    <div className="Profile-main">
      <MyMissions />
      <Myrocket />
    </div>
  );
}

export default MyProfile;
