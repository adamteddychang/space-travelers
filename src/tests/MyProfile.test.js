import React from "react";
import renderer from 'react-test-renderer';
import { Provider } from "react-redux";
import  store  from "../Redux/configureStore";
import MyProfile from "../Components/MyProfile/MyProfile";



describe('MyProfile Component test', ()=>{
  it ("Renders MyProfile",()=>{
    const app = renderer.create(
      <Provider store={store}>
        <MyProfile />

      </Provider>
    ).toJSON();
    expect(app).toMatchSnapshot();
  })
})