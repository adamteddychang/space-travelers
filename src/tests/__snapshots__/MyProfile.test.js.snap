// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyProfile Component test Renders MyProfile 1`] = `
<div
  className="Profile-main"
>
  <div
    className="My-Missions"
  >
    <h2>
      My Missions
    </h2>
    <ul
      className="ul-contain-miss"
    >
      <li>
        Empty
      </li>
    </ul>
  </div>
  <div
    className="My-Rockets"
  >
    <h2>
      My Rockets
    </h2>
    <p>
      Empty
    </p>
  </div>
</div>
`;
